app/javascript/controllers/category_filter_controller.js

```
@@ -0,0 +1,209 @@
+
+  // Keyboard navigation support
+  handleKeydown(event) {
+    if (event.target === this.searchInputTarget) {
+      switch (event.key) {
+        case 'Escape':
+          this.clearSearch()
+          break
+        case 'Enter':
+          event.preventDefault()
+          // Focus first visible category link
+          const firstVisible = this.categoriesListTarget.querySelector('.category-item[style*="display: block"] a, .category-item:not([style*="display: none"]) a')
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)